Framework/Muon/inc/MantidMuon/MuonPairingAsymmetry.h

```diff
@@ -0,0 +1,70 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#ifndef MANTID_MUON_MUONPAIRINGASYMMETRY_H_
+#define MANTID_MUON_MUONPAIRINGASYMMETRY_H_
+
+#include "MantidAPI/Algorithm.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspaceGroup.h"
+
+using namespace Mantid::API;
+
+namespace Mantid {
+namespace Muon {
+
+class DLLExport MuonPairingAsymmetry : public API::Algorithm {
+public:
+  MuonPairingAsymmetry() : API::Algorithm() {}
+  ~MuonPairingAsymmetry() {}
+
+  const std::string name() const override { return "MuonPairingAsymmetry"; }
+  int version() const override { return (1); }
+  const std::string category() const override { return "Muon\\DataHandling"; }
+  const std::string summary() const override {
+    return "Apply a pairing asymmetry calculation between two detector groups "
+           "from Muon data.";
+  }
+  const std::vector<std::string> seeAlso() const override {
+    return {"MuonProcess"};
```

Comment:
we should have muonpreprocess, muonGroupingCounts and muonGroupingAsymmetry and child algs.